engineapi/proxy.go

```diff
@@ -172,62 +425,241 @@ type Proxy struct {
 
 	grpcClient *imclient.ProxyClient
 
-	lock *sync.RWMutex
+	lock sync.RWMutex
+
+	inCall     int
+	inCallLock sync.RWMutex
+
+	nextResync time.Time
+	resyncLock sync.RWMutex
+
+	nextGC  *time.Time
+	gcRetry int
+	gcLock  sync.RWMutex
+}
+
+type EngineClientProxy interface {
+	EngineClient
+	BackupTargetBinaryClient
+
+	getEngineClientProxy() engineClientProxy
+
+	calling()
+	Done()
+}
+
+func (p *Proxy) getEngineClientProxy() engineClientProxy {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	return p
+}
```

Comment:
Why we need this method? actually this just return the same object?

Reply:
Returning `engineClientProxy` interface.